enterprise/server/backends/userdb/userdb.go

```diff
@@ -233,30 +241,144 @@ func (d *UserDB) CreateAPIKey(ctx context.Context, groupID string, label string,
 		return nil, err
 	}
 
-	return createAPIKey(d.h.DB(ctx), groupID, newAPIKeyToken(), label, caps, visibleToDevelopers)
+	return createAPIKey(d.h.DB(ctx), "" /*=userID*/, groupID, newAPIKeyToken(), label, caps, visibleToDevelopers)
 }
 
 func (d *UserDB) GetUserAPIKeys(ctx context.Context, groupID string) ([]*tables.APIKey, error) {
-	return nil, status.UnimplementedError("Not implemented")
+	if groupID == "" {
+		return nil, status.InvalidArgumentError("Group ID cannot be empty.")
+	}
+	u, err := perms.AuthenticatedUser(ctx, d.env)
+	if err != nil {
+		return nil, err
+	}
+	if err := perms.AuthorizeGroupAccess(ctx, d.env, groupID); err != nil {
+		return nil, err
+	}
+	q := query_builder.NewQuery(`SELECT * FROM APIKeys`)
+	q.AddWhereClause(`user_id = ?`, u.GetUserID())
+	q.AddWhereClause(`group_id = ?`, groupID)
+	q.SetOrderBy("label", true /*ascending*/)
+	queryStr, args := q.Build()
+	var keys []*tables.APIKey
+	err = d.h.DB(ctx).Transaction(func(tx *db.DB) error {
```

Comment:
Is a transaction needed here? There are no writes involved.

Reply:
Nope - removed it